### PR TITLE
Document how to use tags and branches when importing projects

### PR DESCRIPTION
**Pull request by bridadan**
This corresponds with the changes that will be part of v1.10.0 of Mbed CLI: https://github.com/ARMmbed/mbed-cli/pull/876

FYI @adbridge @ashok-rao

### CODE REVIEW

**Review by bulislaw — APPROVED** [review]

File: docs/tools/CLI/cli-create.md [hunk 1]
```diff
@@ -108,6 +108,29 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+You can import an existing program by using the `mbed import` command.
+
+```
+$ mbed import https://github.com/ARMmbed/mbed-os-example-blinky#mbed-os-5.11.0
+[mbed] Working path "C:\dev" (directory)
+[mbed] Importing program "mbed-os-example-blinky" from "https://github.com/ARMmbed/mbed-os-example-blinky" at branch/tag "mbed-os-5.11.0"
+[mbed] Adding library "mbed-os" from "https://github.com/ARMmbed/mbed-os" at rev #6a0a86538c0b
+```
+
+You can change what version of the program is imported by changing what comes after the `#` character. You may supply a commit hash, a branch name, or a tag name. If you do not provide any of these (nor the `#` character), the latest commit on the `master` branch is imported.
```
Comment (iriark01) [inline docs/tools/CLI/cli-create.md]:
```suggestion
You can specify which version to import using `#` followed by a commit hash, a branch name, or a tag name. If you do not provide any of these (nor the `#` character), the latest commit on the `master` branch will be imported.
```

File: docs/tools/CLI/cli-create.md [hunk 2]
```diff
@@ -108,6 +108,29 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+
+A project is named after the last part of the URL by default. In the above example, a directory named `mbed-os-example-blinky` will be created to contain the imported program. If you wish to change the name of the project while importing, you can do this by supplying it as an extra positional argument:
```
Comment (iriark01) [inline docs/tools/CLI/cli-create.md]:
```suggestion
A project's default name is the last part of the URL (excluding `#` and its value). In the example above, the imported program's project folder is `mbed-os-example-blinky`. To specify a different name, supply it as an extra positional argument in the import command. For example, to name your project `my-blinky`, run:
```

File: docs/tools/CLI/cli-create.md [hunk 3]
```diff
@@ -108,6 +108,29 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+
+```
+$ mbed import https://github.com/ARMmbed/mbed-os-example-blinky#mbed-os-5.11.0 my-blinky
+```
+
+In this case, a directory named `my-blinky` will be created instead.
```
Comment (iriark01) [inline docs/tools/CLI/cli-create.md]: Remove this line if you accept the suggestion above.

File: docs/tools/CLI/cli-create.md [hunk 4]
```diff
@@ -108,6 +108,29 @@ $ mbed compile -t ARM -m K64F --source project2 --source mbed-os --build BUILD/p
...
+
+Running `mbed import` within an existing program will result in an error. If you wish to add a library to an existing project, you can use the `mbed add` command in a similar manner.
```
Comment (iriark01) [inline docs/tools/CLI/cli-create.md]:
```suggestion
<span class="tips">**Tip**: Running `mbed import` within an existing program will result in an error. To add a library to an existing project, use the `mbed add` command.</span>
```

Comment (iriark01) [reply docs/tools/CLI/cli-create.md]: So... we don't have `mbed add` documented, I think

**Review by iriark01 — CHANGES_REQUESTED** [review]
Love your work, Brian.

File: docs/tools/CLI/cli-create.md (hunk 3 above)
Comment (bridadan) [reply docs/tools/CLI/cli-create.md]: It does appear to be missing, yes. That's a shame, perhaps I could jot something down quickly in this document.

File: docs/tools/CLI/cli-create.md (hunk 4 above)
Comment (bridadan) [reply docs/tools/CLI/cli-create.md]: It does appear to be missing, yes. That's a shame, perhaps I could jot something down quickly in this document.

Comment (iriark01) [reply docs/tools/CLI/cli-create.md]: Go on, then. Go go go!

**Comment by iriark01** [comment]
Merging. Brian will open a new PR for moving `mbed add` and related content.

**Pull request merged**